File: achiveserver/src/main/java/com/example/achiveserver/entity/Achiv.java
package com.example.achiveserver.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "achiv", schema = "achiv")
public class Achiv {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "class", length = 100)
    private String classField;

    @Column(name = "description", nullable = false, length = 200)
    private String description;

    public Achiv(Integer id, String name, String classField, String description) {
        this.id = id;
        this.name = name;
        this.classField = classField;
        this.description = description;
    }

    public Achiv() {
    }

    public Achiv(Integer id) {
        this.id = id;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getClassField() {
        return classField;
    }

    public void setClassField(String classField) {
        this.classField = classField;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: achievmentsclient/src/app/model/achiv.ts
export class Achiv {
  [key: string]: any
  private _id: number;
  private _name: string;
  private _classField: string;
  private _description: string;
  private _checked: boolean;

  constructor(id: number, name: string, clas: string, description: string, checked: boolean) {
    this._id = id;
    this._name = name;
    this._classField = clas;
    this._description = description;
    this._checked = checked;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get classField(): string {
    return this._classField;
  }

  set classField(value: string) {
    this._classField = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get checked(): boolean {
    return this._checked;
  }

  set checked(value: boolean) {
    this._checked = value;
  }


}
